woocommerce/woocommerce-ios
# Update OrderStats to use Storage fmwk + ResultsController

This PR fixes #258 by doing the following:

* **Networking:** Updated the`OrderStats` & `OrderStatsItem` models — `MIContainer` is no longer stored as a var, instead we now decode the individual fields immediately when the incoming JSON is parsed
* **Storage:** Updated the v4 data model to also contain the `OrderStats` and `OrderStatsItem` entities
* **Storage:** Added `loadOrderStats()` to `StorageType+Extensions`
* **Yosemite:** Added `StorageOrderStats` and `StorageOrderStatsItem` models
* **Yosemite:** Updated `StatsStore` to now upsert `OrderStats` when retrieved from the server
* **Yosemite:** Updated `StatsAction.retrieveOrderStats()` action to no longer return `OrderStats`
* **WooCommerce:** `PeriodDataViewController` now uses a `ResultsController` for the Order data
* **WooCommerce:** Simplified `StoreStatsViewController`'s `syncOrderStats` function
* Updated a bunch of unit tests around `OrderStats`
* Cleaned up a bunch of related doc comments

## Testing

0. Make sure everything builds without issue
1. Verify the unit tests are ✅ 
2. Run the app - make sure everything looks ok with the order fields and chart
3. Close the app, place a new order (or 2) on the store you are logged into, and open the app again (maybe pull to refresh) - verify the fields/chart updates as expected

@jleandroperez Could you take a 👀 at this PR? Thanks!!! 🙇
